R/request_class.R

```diff
@@ -1,65 +1,15 @@
-#' @title A request object
-#' @description object that handled all aspects of a request
-#' @export
-#' @keywords internal
-#' @examples
-#' url <- "https://hb.opencpu.org/post"
-#' body <- list(foo = "bar")
-#' headers <- list(
-#'   `User-Agent` = "libcurl/7.54.0 r-curl/3.2 crul/0.5.2",
-#'   `Accept-Encoding` = "gzip, deflate",
-#'   Accept = "application/json, text/xml, application/xml, */*"
-#' )
-#'
-#' (x <- Request$new("POST", url, body, headers))
-#' x$body
-#' x$method
-#' x$uri
-#' x$host
-#' x$path
-#' x$headers
-Request <- R6::R6Class(
-  'Request',
-  public = list(
-    #' @field method (character) http method
-    method = NULL,
-    #' @field uri (character) a uri
-    uri = NULL,
-    #' @field body (character) named list
-    body = NULL,
-    #' @field headers (character) named list
-    headers = NULL,
-    #' @field disk (logical) xx
-    disk = NULL,
-    #' @field fields (various) request body details
-    fields = NULL,
-    #' @field output (various) request output details, disk, memory, etc
-    output = NULL,
-    #' @field policies (various) http policies, used in httr2 only
-    policies = NULL,
+vcr_request <- function(method = "GET", uri, body = NULL, headers = list()) {
```

Comment:
isn't it bad practice to have a param with no default after a param with a default?